castai/types.go

```diff
@@ -187,6 +187,7 @@ func (c *ChartSource) Validate() error {
 }
 
 type AKSInitDataRequest struct {
-	CloudConfigBase64       string `json:"cloudConfigBase64"`
-	ProtectedSettingsBase64 string `json:"protectedSettingsBase64"`
+	CloudConfigBase64       string  `json:"cloudConfigBase64"`
+	ProtectedSettingsBase64 string  `json:"protectedSettingsBase64"`
+	Architecture            *string `json:"architecture,omitempty"`
```

Comment:
why its nullable?

Reply:
IMO Everything new should be optional. What is the concern?

Reply:
Just added as suggested